### DISCUSSION TITLE
How to filter() jsonb array with javascript sdk?

### DISCUSSION THREAD

**Discussion by taufik-nurhidayat** [supabase/supabase#8159] [Questions] [answered]
This is my code in frontend
```javascript
const { data, error } = await supabase.from('posts').select('*').filter('categories', 'cs', JSON.stringify({"slug": slug}))
```
it is my data in categories column
```json
[ {"slug": "tutorial", "id": "1", "name": "Tutorial"}]
```
And its return data
```json
{"code":"22P02","details":"Unexpected array element.","hint":null,"message":"malformed array literal: \"{\"slug\":\"tutorial\"}\""}
```
I dont know how to describe it, but i want to pass an filter function

**Comment by GaryAustin1** [comment, marked as answer]
You can filter on a jsonb column, but not on array column (other than I think selecting particular index into the array, which is not what you want).  You will need to use rpc with a postgres function to do a filter on an array of jsonb.

Edit: Here was a suggestion on using a computed column to search from the client side :https://github.com/supabase/supabase/discussions/2415

**Reply by taufik-nurhidayat** [reply]
well, i make new collum array: text[]. Now i have 2 collum array, first collum is jsonb array and second collum is text array used for filter only. That's work for me.
